### ISSUE TITLE
Allow output bucket specification / minio egress

### ISSUE THREAD

**Issue by DSchmidtDev** [pachyderm/pachyderm#2868]
**What is the ask?**

I think it would be useful to allow the specification of an output bucket to separate the pipeline name and the data-repository handling.

An alternative would be to support _minio_ as an egress destination.

**What is the goal / desired outcome?**

So that I can accomplish my goal to save results of my pipeline in a separate bucket so that other users can have access to it.

**If there is a way to accomplish this today via workaround, what does that require?**

**Comment by David-Development** [comment]
I would like to add that the current "error" message is kind of difficult to interpret. I'm running a On-Premise installation of Pachyderm (using Minio as an S3 Storage). Setting the egress output to `"URL": "s3://my-bucket/pachyderm-graph/"` results in a failing job. When inspecting the job I can see the following Reason: `amazon-region not found`. So if I understand this right, using minio as an output isn't supported right now? 

I'm using the following command to deploy Pachyderm (using minio):
```bash
pachctl deploy custom --namespace pachyderm --persistent-disk google --object-store s3 pachyderm-s3 10 pachyderm-bucket ${S3_STORAGE_USERNAME} ${S3_STORAGE_PASSWORD} ${S3_STORAGE_HOST} --dynamic-etcd-nodes 1 --dry-run > deployment.json
```

**Comment by David-Development** [comment]
@marcadella Indeed, I guess I mixed them up a little. Thank you for the clarification! Is there any documentation on the difference between the two? I couldn't find any documentation on how to define an external object store. Therefore I assumed, that pachyderm uses the "internal" one by default. 

@JoeyZwicker Maybe it would be helpful to include this kind of information in the documentation? The current docs on egress only mentions that you can push results to an external provider ([Link](https://docs.pachyderm.io/en/latest/fundamentals/getting_data_out_of_pachyderm.html#exporting-data-via-egress)). However the [pipeline specs](https://docs.pachyderm.io/en/latest/reference/pipeline_spec.html) only includes the bucket name and directory (see example below). Therefore I assumed, that by default it will use the internal s3 storage.

```
"egress": {
    "URL": "s3://bucket/dir"
},
```

I can only agree with @marcadella that it would be tremendously useful the use minio as an egress storage.

**Comment by marcadella** [comment]
To use an egress AWS storage you would do the following:
1. Run `pachctl deploy storage amazon <region> <id> <secret>`
This creates a Kubernetes secret containing your amazon credentials
2. In your pipeline  you add
```
"egress": {
    "URL": "s3://bucket/dir"
}
```
Unfortunately for us, when Pachyderm sees `s3`, it makes up an endpoint that suits AWS (something like `s3.<location>.amazon.com`). If we had a way to set-up the endpoint when we do `pachctl deploy storage` (by adding a backend `s3` for instance that accepts an endpoint), we would be able to egress to Minio. But for now this is not possible.

**Comment by gabrielgrant** [comment]
agreed, that seems like i reasonable ask. I believe Minio egress is something we're going to need for the on-prem OpenShift deployment we're currently working on

**Comment by Nick-Harvey** [comment]
Adding the docs label to clarify our docs a bit better as requested by @David-Development

**Comment by jdoliner** [comment]
We should probably accept something like:

```
"egress": {
    "URL": "s3api://<endpoint>/bucket/dir"
}
```

Specifying the endpoint in the storage secret will make it hard to egress to multiple endpoints, and just moves the information farther away from where it's actually used which will make it harder to understand / audit.
